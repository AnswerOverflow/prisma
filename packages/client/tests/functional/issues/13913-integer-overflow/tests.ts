import { get } from '../../../../../../helpers/blaze/get'
import testMatrix from './_matrix'
// @ts-ignore
import type { PrismaClient } from './node_modules/@prisma/client'

declare let prisma: PrismaClient

const id = 'ac6be4a35f2ed6e94bb2928d'

testMatrix.setupTestSuite(({ provider }) => {
  test('int overflow', async () => {
    const result = prisma.resource.create({
      data: {
        id,
        number: 2265000000,
      },
    })

    const success = () => expect(result).resolves.toMatchSnapshot()
    const failure = () => expect(result).rejects.toMatchPrismaErrorSnapshot()

    await get(
      {
        postgresql: success,
        cockroachdb: success,
        mongodb: success,
        mysql: failure,
        sqlite: failure,
        sqlserver: failure,
      },
      provider,
    )()
  })
})
