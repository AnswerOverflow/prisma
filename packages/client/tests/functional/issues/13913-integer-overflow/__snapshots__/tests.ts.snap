// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issues.13913-integer-overflow (provider=cockroachdb) int overflow 1`] = `
Object {
  id: ac6be4a35f2ed6e94bb2928d,
  number: -2029967296,
}
`;

exports[`issues.13913-integer-overflow (provider=mongodb) int overflow 1`] = `
Object {
  id: ac6be4a35f2ed6e94bb2928d,
  number: 2265000000,
}
`;

exports[`issues.13913-integer-overflow (provider=mysql) int overflow 1`] = `

Invalid \`prisma.resource.create()\` invocation in
/client/tests/functional/issues/13913-integer-overflow/tests.ts:0:0

   XX 
  XX testMatrix.setupTestSuite(({ provider }) => {
  XX   test('int overflow', async () => {
→ XX     const result = prisma.resource.create(
Value out of range for the type. Out of range value for column 'number' at row 1
`;

exports[`issues.13913-integer-overflow (provider=postgresql) int overflow 1`] = `
Object {
  id: ac6be4a35f2ed6e94bb2928d,
  number: -2029967296,
}
`;

exports[`issues.13913-integer-overflow (provider=sqlite) int overflow 1`] = `

Invalid \`prisma.resource.create()\` invocation in
/client/tests/functional/issues/13913-integer-overflow/tests.ts:0:0

   XX 
  XX testMatrix.setupTestSuite(({ provider }) => {
  XX   test('int overflow', async () => {
→ XX     const result = prisma.resource.create(
Inconsistent column data: Conversion failed: Value 2265000000 does not fit in an INT column, try migrating the 'number' column type to BIGINT
`;

exports[`issues.13913-integer-overflow (provider=sqlserver) int overflow 1`] = `

Invalid \`prisma.resource.create()\` invocation in
/client/tests/functional/issues/13913-integer-overflow/tests.ts:0:0

   XX 
  XX testMatrix.setupTestSuite(({ provider }) => {
  XX   test('int overflow', async () => {
→ XX     const result = prisma.resource.create(
Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(TokenError { code: 8115, state: 2, class: 16, message: "Arithmetic overflow error converting expression to data type int.", server: "xxxxxxxxxxxx", procedure: "", line: 1 }) })
`;
